### PR TITLE
fix(iOS): fire onMomentumScrollEnd when UIScrollView is removed from window

### PR DESCRIPTION
**Pull request by shubhamguptadream11**
## Summary:

Solves this issue: https://github.com/facebook/react-native/issues/46276

## Changelog:

[IOS] [ADDED] - fire onMomentumScrollEnd when UIScrollView is removed from window

**Why the issue is happening?**
The `onMomentumScrollEnd` event is typically triggered by the `UIScrollView` delegate methods `scrollViewDidEndDecelerating` and `scrollViewDidEndScrollingAnimation`. However, if the scroll view is removed from the window while navigating away, these delegate methods are not called, resulting in the event not being dispatched.

This behaviour was particularly problematic in scenarios where a scroll view is in motion, and the user navigates away from the screen before the scrolling completes. In such cases, the `onMomentumScrollEnd` event would never fire, which further make scroll area un touchable or un responsive.

**What we changed?**
In the didMoveToWindow method, we added logic to handle the scenario where the UIScrollView is being removed from the window (i.e., when the component is unmounted or the user navigates away). Here’s a breakdown of the changes:

- **Added a Check for Scroll State:** We check if the UIScrollView was decelerating or had stopped tracking (_scrollView.isDecelerating || _scrollView.isTracking == NO).

- **Manually Triggered onMomentumScrollEnd:** If the scroll view was in motion and is being removed from the window, we manually trigger the `onMomentumScrollEnd` event to ensure that the final scroll state is captured.

**_I had fixed this issue on both Old and New arch._**

## Test Plan:

Attaching a video with working solution:

https://github.com/user-attachments/assets/1a1f3765-3f11-46c3-af18-330c88478db8

### CODE REVIEW

**Comment by shubhamguptadream11** [comment]
Hi @cipolleschi,
I hope you're doing well. Could you please take a look and review this PR.
I had added a fix for Old and New both.
Thanks!

File: packages/react-native/React/Views/ScrollView/RCTScrollView.m [hunk 1]
```diff
@@ -848,6 +848,18 @@ - (void)scrollViewDidEndZooming:(UIScrollView *)scrollView withView:(UIView *)vi
   RCT_FORWARD_SCROLL_EVENT(scrollViewDidEndZooming : scrollView withView : view atScale : scale);
 }
 
+- (void)didMoveToWindow {
+  [super didMoveToWindow];
+  if (self.window == nil) {
+    // Check if the ScrollView was in motion
+    if (_scrollView.isDecelerating || _scrollView.isTracking == NO) {
```
Comment (cipolleschi) [inline packages/react-native/React/Views/ScrollView/RCTScrollView.m:855]: Nit for consistency:
```suggestion
    if (_scrollView.isDecelerating || !_scrollView.isTracking) {
```

**Review by cipolleschi — APPROVED** [review]
Thanks for looking into this!

File: packages/react-native/React/Views/ScrollView/RCTScrollView.m (hunk 1 above)
Comment (shubhamguptadream11) [reply packages/react-native/React/Views/ScrollView/RCTScrollView.m]: Changes done.

**Comment by shubhamguptadream11** [comment]
Hey @cipolleschi ,
Do you think it would be possible to backport this to version 0.72.5 at least, since we need to use it on that version? If not, maybe backporting to 0.73 would also work.

Thanks in advance!

**Comment by cipolleschi** [comment]
0.72 is out of support, so no, sadly.
We can backport it to 0.73, for sure.

**Comment by shubhamguptadream11** [comment]
@cipolleschi  That's fantastic! Supporting this up until version 0.73 sounds like a solid plan. If there's anything I can do to assist or contribute in any way, I am happy to help!!

**Comment by cipolleschi** [comment]
I already opened cherry pick requests for 0.76. See https://github.com/reactwg/react-native-releases/issues/510 and https://github.com/reactwg/react-native-releases/issues/511
Can you open the same for 0.74 and 0.73?

**Comment by shubhamguptadream11** [comment]
@cipolleschi I'm not familiar with the exact process. Could you kindly guide me through the steps involved in doing this?

**Comment by shubhamguptadream11** [comment]
@cipolleschi 

Just following up on my previous message. I understand you’re likely busy, but I was hoping to get your guidance on the steps for this process whenever you have a moment.

**Pull request closed without merging**